archive/dug_6/dug_6_dasharo_configuration_utility_status_update.md

```diff
@@ -0,0 +1,257 @@
+    across multiple systems, ensuring consistency in deployment.
+- #### Sane Defaults Verification
+  - Verifying and aligning user configurations with established sane defaults,
+    ensuring optimal security and performance settings for various hardware
+    setups 
+- #### Data Migration Tools
+  - Smooth transition of data and settings from proprietary BIOS to Dasharo
+  firmware.
+- #### Enterprise Security Features
+  - Advanced security features like TPM configuration, secure boot
+  customization, and hardware-based encryption support.
+- #### Integration of tools like Chipsec and Binarly for security audits
+
+---
+
+# Current Status
+
+- ~~Custom Configuration Profiles~~
+- ~~Sane Defaults Verification~~
+- ~~Data Migration Tools~~
+- ~~Enterprise Security Features~~
+- ~~Chipsec and HSI Checks Integration~~
+- ~~Integration with Commercial Tools like Binarly~~
+
+<br>
+
+.center.image-35[![](/img/slacking_off.png)]
+
+???
+
+- And here is what we actually delivered.
+- But that doesn't mean we've been inactive.
+
+---
+
+# The problem
+
+.center[
+<img src="/img/x86_ring_level_arrow.svg" height="550px" style="margin-top:-70px; margin-left:20px">
+]
+
+???
+
+- _SMM is a special operating mode in x86 processors that handles system-wide
+functions like:_
+  - _power management_
+  - _system hardware control_
+  - _firmware settings storage_
+- This separates this funtionality from the userspace protecting it from
+  software attacks
+- Coreboot stores it's firmware settings in the SMMS
+  - Until recently they could be only modified by hand from the UEFI menu
+  - So we made a tool for changing the settings via an util/smmstoretool
+- We had to develop a way to access the contents of the dasharo binary without
+  manually setting the UEFI variables.
+- Fun fact: so new the documentation still wasn't merged
+
+---
+
+# Our solution
+
+.center[
+<img src="/img/coreboot_repo_smmstoretool_sc.png" height="450px" style="margin-left:50px">
+]
+
+???
+
+- Here at 3mdeb we've developed and upstreamed the `smmstoretool` utility for
+  offline modification of the SMM storage in the coreboot or Dasharo binaries
+- In other words we can now set our firmware settings without booting it up
+
+---
+
+# What does this mean to Dasharo users
+
+<!-- Accepted values are slanted so they appear correctly on the presentation -->
+
+.code-15px[
+#### Get a list of settings in a binary
+```sh
+$ ./dcu variable --list coreboot.rom
+Settings in coreboot.rom:
+NAME		VALUE		    ACCEPTED VALUES
+MeMode	  Enabled          Enabled / Disabled (Soft) / Disabled (HAP)
+```
+#### Change a setting
+```sh
+$ ./dcu variable coreboot.rom --set "MeMode" --value "Disabled (Soft)"
+```
+```sh
+$ ./dcu variable --list coreboot.rom
+Settings in coreboot.rom:
+NAME		VALUE			ACCEPTED VALUES
+MeMode	  Disabled (Soft)  Enabled / Disabled (Soft) / Disabled (HAP)
+```
+]
+
+???
+
+- We expanded the DCU with additional utilities
+
+TODO how exactly DCU maps SMM settings
```

Comment:
We simply call smmstore in DCU.
How smmstore tool does it? You do not need to know, if needed we can open the source code.

Reply:
Ok. Won't cover this then